dali/tensorflow/daliop.cc

```diff
@@ -60,20 +61,20 @@ tf::TensorShape DaliToShape(int64_t* ns) {
 
 REGISTER_OP("Dali")
   .Attr("serialized_pipeline: string")
-  .Attr("shape: shape")
+  .Attr("shapes: list(shape) >= 1")
   .Attr("num_threads: int = -1")
   .Attr("device_id: int = -1")
-  .Attr("image_type: {float, int32, int64, half} = DT_FLOAT")
-  .Attr("label_type: {float, int32, int64} = DT_INT64")
+  // .Attr("data_type: {float, int32, int64, half} = DT_FLOAT")
   .Attr("prefetch_queue_depth: int = 2")
-  .Output("batch: image_type")
-  .Output("label: label_type")
+  .Output("data: dtypes")
+  .Attr("dtypes: list({float, int32, int64, half}) >= 1")
   .SetShapeFn([](tf::shape_inference::InferenceContext* c) {
-    tf::PartialTensorShape shape;
-    TF_RETURN_IF_ERROR(c->GetAttr("shape", &shape));
+    // define shape for the first output
+    std::vector<tf::PartialTensorShape> shapes;
+    TF_RETURN_IF_ERROR(c->GetAttr("shapes", &shapes));
     tf::shape_inference::ShapeHandle passed_shape;
     TF_RETURN_IF_ERROR(
-        c->MakeShapeFromPartialTensorShape(shape, &passed_shape));
+        c->MakeShapeFromPartialTensorShape(shapes[0], &passed_shape));
     TF_RETURN_IF_ERROR(
```

Comment:
Static shape inference should be done for each ouput:
```
for (unsigned i = 0; i < shapes.size(); ++i) {
    tf::shape_inference::ShapeHandle passed_shape;
    TF_RETURN_IF_ERROR(
        c->MakeShapeFromPartialTensorShape(shapes[i], &passed_shape));
     c->set_output(i, passed_shape); 
}
```

Reply:
And we should remove 
https://github.com/NVIDIA/DALI/blob/dea4f1a1249cbdcc3e5f0fd200ddd184ad5aad5d/dali/tensorflow/daliop.cc#L60

Reply:
Done